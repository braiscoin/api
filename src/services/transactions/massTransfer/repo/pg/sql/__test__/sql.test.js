const sql = require('..');

const filterValues = {
  recipient: 'recipient',
  assetId: 'assetId',
};

test('Sql search supports type-specific filters', () => {
  expect(sql.default.search(filterValues)).toMatchSnapshot();
});
