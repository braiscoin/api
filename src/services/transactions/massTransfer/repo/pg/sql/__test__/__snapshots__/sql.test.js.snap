// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sql search supports type-specific filters 1`] = `"select \\"t\\".\\"uid\\" as \\"uid\\", \\"t\\".\\"id\\" as \\"id\\", \\"t\\".\\"fee\\" as \\"fee\\", \\"t\\".\\"height\\" as \\"height\\", \\"t\\".\\"tx_type\\" as \\"tx_type\\", \\"t\\".\\"time_stamp\\" as \\"time_stamp\\", \\"t\\".\\"signature\\" as \\"signature\\", \\"t\\".\\"proofs\\" as \\"proofs\\", \\"t\\".\\"tx_version\\" as \\"tx_version\\", \\"t\\".\\"status\\" as \\"status\\", \\"t\\".\\"sender\\" as \\"sender\\", \\"t\\".\\"sender_public_key\\" as \\"sender_public_key\\", \\"t\\".\\"asset_id\\" as \\"asset_id\\", \\"t\\".\\"attachment\\" as \\"attachment\\", \\"tfs\\".\\"recipient_alias\\" as \\"recipient_alias\\", \\"tfs\\".\\"recipient_address\\" as \\"recipient_address\\", \\"tfs\\".\\"amount\\" as \\"amount\\", \\"tfs\\".\\"position_in_tx\\" as \\"position_in_tx\\" from \\"txs_11\\" as \\"t\\" left join \\"txs_11_transfers\\" as \\"tfs\\" on \\"tfs\\".\\"tx_uid\\" = \\"t\\".\\"uid\\" where \\"t\\".\\"uid\\" in (select \\"t\\".\\"uid\\" from \\"txs_11\\" as \\"t\\" left join \\"txs_11_transfers\\" as \\"tfs\\" on \\"tfs\\".\\"tx_uid\\" = \\"t\\".\\"uid\\" where \\"asset_id\\" = 'assetId' and tfs.recipient_address = coalesce((select sender from txs_10 where alias = 'recipient' limit 1), 'recipient') group by \\"t\\".\\"uid\\")"`;
